webapp/src/main/java/com/box/l10n/mojito/service/security/user/UserService.java

```diff
@@ -385,4 +405,80 @@ public Page<User> findByUsernameOrName(String username, String search, Pageable
         });
     return users;
   }
+
+  /**
+   * Gets a service by name and if it doesn't exist create a created user.
+   *
+   * <p>All admin service accounts should already be created beforehand however, create a non-admin
+   * account to so long. If they need admin permissions, we will promote the account
+   *
+   * @param serviceName
+   * @return
+   */
+  public User getServiceAccountUser(String serviceName) {
+    if (serviceDisambiguator == null) {
+      logger.debug("Service Disambiguator is null. Falling back to regular exact match logic");
+      return userRepository.findByUsername(serviceName);
+    }
+
+    Optional<List<User>> users =
+        userRepository.findService(serviceName, headerSecurityConfig.servicePrefix);
+    if (users.isEmpty()) {
+      logger.debug("No matching services found as per DB query");
+      sendPagerDutyNotification(serviceName);
+      logger.error(
+          "Service '{}' attempted and failed authentication. No matching services found.",
+          serviceName);
+      return null;
+    }
+
+    User matchingUser =
+        serviceDisambiguator.findServiceWithCommonAncestor(users.get(), serviceName);
+    if (matchingUser == null) {
+      logger.debug("No matching services found as per ServiceDisambiguator");
+      sendPagerDutyNotification(serviceName);
```

Comment:
The sending of a PD notification should be configurable here, some deployments may not wish to raise an incident for this particular scenario

Reply:
That's fair. I've updated this